example-project/project.json

```diff
@@ -0,0 +1,168 @@
+{
+  "sketches": {
+    "c64ceed4673": {
+      "id": "c64ceed4673",
+      "moduleId": "solid",
+      "title": "Solid",
+      "paramIds": [
+        "64ceed46737",
```

Comment:
A few thoughts on the file format

1. Do we need id when we can just use the objects key?
2. Will paramIds ever be shared? Kind of feel like the file would be more readable if this was just a `"params"` array, and the param objects (starting on line 37) also just contained their value

I'm assuming that this current layout is close to how it is stored in memory? Think it could be worth having a de/serialize step so the file could be more readable in commits, or if they ever need to be touched by hand

Reply:
You're right in thinking this is how it's stored in the store. I'm a bit scared to have an extra step when saving/loading, it feels like a lot of extra complexity. 

The reason for the IDs in the object is just out of convenience, sometimes you've passed the param around in a function or a component and you need to still get the id later. 

As for having the params nested rather than flat, I see how this would be more readable but I think it could be quite tricky trying to get this to work. Your example would probably be possible but once we have inputs, would you then want those nested inside the params? 😅 What if at some point we want two things to share a node, do we copy the node twice?

I can imagine this getting out of hand, with dependency loops, etc. There is a good reason to keeping the state flat in the store and I think this reason extends to the file format.

However, there is one small thing we can easily do to improve readability: prefix the node IDs with `param_`, `input_`, etc.

Reply:
Good point on the id's, makes a lot of sense as an in-memory object.

I also wasn't sure if you were thinking about shared nodes, I'm into it! Makes it easy to treat an object as a clone vs a separate instance for example.

I don't think we need a prefix for the node, you can get the info you need via the type param.

I do still wonder about the nodeValue in the node, but it's not too big of a deal